CMakeLists.txt

```diff
@@ -13,30 +8,33 @@ mark_as_advanced(VERSION_MAJOR VERSION_MINOR VERSION_PATCH COSE_VERSION)
 
 project ("cose-c" VERSION "${COSE_VERSION}")
 
-### setup options
-option (use_context	"Use context pointer for COSE functions" ON)
-option (verbose         "Produce verbose makefile output" OFF)
-option (optimize        "Optimize for size" OFF)
-option (fatal_warnings  "Treat build warnings as error" OFF)
-option (coveralls       "Generate coveralls data" ON)
-option ( coveralls_send "Send data to coveralls site" OFF )
-option (COSE_C_BUILD_DOCS      "Create docs using Doxygen" OFF )
-option (COSE_C_BUILD_TESTS      "build tests" ON )
-option (COSE_C_BUILD_DUMPER      "build dumper" ON )
-option (BUILD_SHARED_LIBS "Build Shared Libraries" ON)
-option (COSE_C_USE_MBEDTLS    "Use MBedTLS for the Crypto Package" OFF)
-option(COSE_C_USE_FIND_PACKAGE "Use cmake find_package instead of using cmake project_add" OFF)
-option (include_encrypt  "Include COSE_ENCRYPT" ON)
-option (include_encrypt0 "Include COSE_ENCRYPT0" ON)
-option (include_mac      "Include COSE_MAC" ON)
-option (include_mac0     "Include COSE_MAC0" ON)
-option (include_sign     "Include COSE_SIGN" ON)
-option (include_sign1    "Include COSE_SIGN1" ON)
-option (include_sign0    "Include COSE_SIGN1" ON)
-
-if (NOT include_sign0)
-   message ( "Use include_sign1 rather than include_sign0" )
-   set (include_sign1 ${include_sign0})
+###############################################################################
+# OPTIONS
+###############################################################################
+
+option (COSE_C_USE_CONTEXT	         "Use context pointer for COSE functions" ON)
+option (COSE_C_VERBOSE              "Produce COSE_C_VERBOSE makefile output" OFF)
+option (COSE_C_OPTIMIZE             "Optimize for size" OFF)
+option (COSE_C_FATAL_WARNINGS       "Treat build warnings as error" OFF)
+option (COSE_C_COVERALLS            "Generate coveralls data" ON)
+option (COSE_C_COVERALLS_SEND       "Send data to coveralls site" OFF )
+option (COSE_C_BUILD_DOCS           "Create docs using Doxygen" OFF )
+option (COSE_C_BUILD_TESTS          "build tests" ON )
+option (COSE_C_BUILD_DUMPER         "build dumper" ON )
+option (BUILD_SHARED_LIBS           "Build Shared Libraries" ON)
+option (COSE_C_USE_MBEDTLS          "Use MBedTLS for the Crypto Package" OFF)
```

Comment:
Curious why you did not namespace BUILD_SHARED_LIBS

Reply:
because this variable is a cmake built-in which is passed automatically by cmake and package managers when parent project is shared...

Reply:
Thanks for the info